### PR TITLE
Global design

### PR DESCRIPTION
**Pull request by agnes97**
_Attention! This branch is on homepage branch. Please view it accordingly!_

Add global Page component (e.g. global wrapper)
Add global title
Add Props for CSS that differs in Dark/Light theme
Add global link
Add global enum for Media Queries
Add global paragraph
Add Font sizes and refactor accordingly

### CODE REVIEW

File: src/components/ButtonNav/index.tsx [hunk 1]
```diff
@@ -1,36 +1,53 @@
...
 import { UserEdit, Users } from '@styled-icons/fa-solid'
 import { PermContactCalendar } from '@styled-icons/material-rounded'
+import { Theme, useThemeControl } from '../Theme'
 
 interface Props {
 	login: boolean
 }
 
-const NavBeforeLogin = (
+export const NavBeforeLogin = (theme: Theme): React.ReactElement => (
```
Comment (tomas90hampl) [inline src/components/ButtonNav/index.tsx:11]:
```suggestion
const NavBeforeLogin: React.FC<{theme: Theme}> = ({theme}) => (
```
- For uniform syntax use React.FC and generics type props
- Export is unnecessary here

If reason of this syntax is education, so I understand :)

File: src/components/ButtonNav/index.tsx [hunk 2]
```diff
@@ -1,36 +1,53 @@
...
-		</StyledLink>
-		<StyledLink to="/matching">
-			<StyledButton icon={Users} title="Matching" />
-		</StyledLink>
-		<StyledLink to="/contacts">
-			<StyledButton icon={PermContactCalendar} title="Kontakty" />
-		</StyledLink>
-	</Wrapper>
-)
+export const NavAfterLogin: React.FC = () => {
```
Comment (tomas90hampl) [inline src/components/ButtonNav/index.tsx:26]: Export is unnecessary here, `NavAfterLogin` is used in the same file

File: src/components/ButtonNav/index.tsx [hunk 3]
```diff
@@ -1,36 +1,53 @@
...
+			</ButtonLink>
+		</Wrapper>
+	)
+}
+
+export const ButtonNav: React.FC<Props> = ({ login }) => {
+	const [theme] = useThemeControl()
+
+	// We use two different ways to pass the theme for learning purposes
+	return login ? <NavAfterLogin /> : NavBeforeLogin(theme)
```
Comment (tomas90hampl) [inline src/components/ButtonNav/index.tsx:52]:
```suggestion
	return login ? <NavAfterLogin /> : <NavBeforeLogin theme={theme} />
```
- Uniform syntax for call component

File: src/components/StyledText/StyledLink/styled.ts [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+}
+
+export const StyledLink = styled(Link)<Props>`
+	text-align: center;
+	text-decoration: none;
+	text-transform: uppercase;
+	color: ${({ changeColor }) =>
+		changeColor === 'darkBlue'
+			? ({ theme }) => theme.text.secondary // Yellow color
+			: ({ theme }) => theme.text.link}; // Default Link color
```
Comment (tomas90hampl) [inline src/components/StyledText/StyledLink/styled.ts:15]: This comment I mean `Yellow color` seem unnecessary to me.

- These properties is dynamic.
- If we add new thema or changed old, we must edit this comments too. 
- Comments should not hamper the sustainability of the code

File: src/components/StyledText/StyledTitle/styled.ts [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+	changeColor?: string
+}
+
+export const TitleH1 = styled.h1<Props>`
+	font-size: ${FontSizes.SizeTitleH1};
+	text-align: center;
+	text-transform: uppercase;
+	color: ${({ changeColor }) =>
+		changeColor === 'darkBlue'
+			? ({ theme }) => theme.text.link // Dark blue color
```
Comment (tomas90hampl) [inline src/components/StyledText/StyledTitle/styled.ts:14]: This comment seem unnecessary to me.

- These properties is dynamic.
- If we add new thema or changed old, we must edit this comments too.
- Comments should not hamper the sustainability of the code

File: src/themes/index.ts [hunk 6]
```diff
@@ -1,5 +1,28 @@
...
+
+	// Clip Text Mask
+	SizeClip = '3.7rem',
+
+	// Button Font Sizes
+	ButtonBig = '1.5rem',
+	ButtonDefault = '1.2rem',
+	ButtonSmall = '0.9rem',
+}
+
```
Comment (tomas90hampl) [inline src/themes/index.ts:25]: :+1:

**Comment by MartinaVilimova** [comment]
Very good work 👍

File: src/components/ButtonNav/index.tsx (hunk 1 above)
Comment (macajind) [reply src/components/ButtonNav/index.tsx:11]: Yes, this was show of two different styles how to handle something like this:
1. Make an indipendent component with own hook etc. (easy to move into a separate file)
2. Make a function that returns a component and pass everything as arguments of such function.

In this case I would say that both styles are interchangeable, but it's true, that we should probably stick with one as a convention. :wink:

File: src/components/NavBar/styled.ts [hunk 7]
```diff
@@ -8,22 +10,30 @@ export const Wrapper = styled.header`
...
+	@media (max-width: ${MediaQueries.Ipad}) {
 		grid-template-columns: 1fr;
 		padding-bottom: 1.5em;
 	}
 `
 
-export const StyledLink = styled(Link)`
-	color: ${({ theme }) => theme.text.link};
+type Props = {
+	// "setcolor" doesn't use camelCase due to it being passed on DOM (Link)
```
Comment (macajind) [inline src/components/NavBar/styled.ts:24]: Interesting. :thinking:

File: src/components/Page/index.tsx [hunk 8]
```diff
@@ -0,0 +1,8 @@
+import React from 'react'
+import { GlobalWrapper } from './styled'
+
+export const Page: React.FC = ({ children }) => {
```
Comment (macajind) [inline src/components/Page/index.tsx:4]: Don't need `export` here, when you export it once again below. :grin:
```suggestion
const Page: React.FC = ({ children }) => {
```

File: src/components/StyledText/StyledParagraph/styled.ts [hunk 9]
```diff
@@ -0,0 +1,9 @@
+import styled from 'styled-components'
+import { FontSizes } from '../../../themes'
+
+export const StyledParagraph = styled.p`
+	font-size: ${FontSizes.SizeDefault};
+	line-height: 1.5em;
+	text-align: left;
```
Comment (macajind) [inline src/components/StyledText/StyledParagraph/styled.ts:7]: Isn't this default? :thinking:

File: src/pages/Login.tsx [hunk 10]
```diff
@@ -2,19 +2,22 @@ import React from 'react'
...
-import LoginForm from '../components/LoginForm'
-import NavBar from '../components/NavBar'
 import Home from './Home'
+import NavBar from '../components/NavBar'
+import Page from '../components/Page'
+import LoginForm from '../components/LoginForm'
 import Footer from '../components/Footer'
 
 const Login: React.FC = () => {
 	return (
```
Comment (macajind) [inline src/pages/Login.tsx:12]: ~~`return`~~ :grin:

File: src/pages/Login.tsx [hunk 11]
```diff
@@ -2,19 +2,22 @@ import React from 'react'
 import { Switch, Route } from 'react-router-dom'
 
 // Components
```
Comment (macajind) [inline src/pages/Login.tsx:4]: Is this comment necessary though? :thinking:

File: src/pages/Matching.tsx [hunk 12]
```diff
@@ -1,12 +1,20 @@
 import React from 'react'
+
+// Components
```
Comment (macajind) [inline src/pages/Matching.tsx:3]: Is this comment necessary though? :thinking:

File: src/pages/Profile.tsx [hunk 13]
```diff
@@ -1,9 +1,12 @@
 import React, { useEffect, useState } from 'react'
-import Footer from '../components/Footer'
-import ContentProfile from '../components/Profile'
-import NavBar from '../components/NavBar'
 import { getProfileData, RandomUser } from '../utils/getUserData'
 
+// Components
```
Comment (macajind) [inline src/pages/Profile.tsx:4]: Is this comment necessary though? :thinking:

File: src/pages/Register.tsx [hunk 14]
```diff
@@ -3,6 +3,7 @@ import React, { useState } from 'react'
 // Components
```
Comment (macajind) [inline src/pages/Register.tsx:3]: Is this comment necessary though? :thinking:

**Review by macajind — COMMENTED** [review]
Overall I like this change very much! :top:

File: src/components/ButtonNav/index.tsx (hunk 1 above)
Comment (agnes97) [reply src/components/ButtonNav/index.tsx:11]: Fixed. :) I kept a function that returns a component and passes everything as an argument of itself. The reason being we usually keep components in separate files already so if they're in one, convention changes and it seems more "clean" to pass an argument.

File: src/components/StyledText/StyledParagraph/styled.ts (hunk 9 above)
Comment (agnes97) [reply src/components/StyledText/StyledParagraph/styled.ts:7]: It is. :sweat_smile: Fixed!

File: src/components/Profile/styled.ts [hunk 15]
```diff
@@ -1,45 +1,38 @@
...
-	font-size: 1.4em;
+	font-size: ${FontSizes.SizeDefault};
 	text-align: center;
 `
 
-export const Description = styled.pre`
-	text-align: justify;
-	white-space: pre-wrap;
-	word-break: break-all;
+export const Description = styled(StyledParagraph)`
```
[the remark this answers is not in the document]
Comment (agnes97) [reply src/components/Profile/styled.ts:31]: I've fixed this, but I also created a separate issue for additional questions. :3

**Pull request merged**